sdk/storage/azure-storage-blob/azure/storage/blob/aio/_blob_service_client_async.py

```diff
@@ -568,14 +569,14 @@ async def undelete_container(self, deleted_container_name, deleted_container_ver
             Specifies the name of the deleted container to restore.
         :param str deleted_container_version:
             Specifies the version of the deleted container to restore.
-        :keyword str new_name:
-            The new name for the deleted container to be restored to.
-            If not specified deleted_container_name will be used as the restored container name.
         :keyword int timeout:
             The timeout parameter is expressed in seconds.
         :rtype: ~azure.storage.blob.aio.ContainerClient
         """
         new_name = kwargs.pop('new_name', None)
+        if new_name:
+            warnings.simplefilter('always', DeprecationWarning)
```

Comment:
Hi Anna! I think we should let the user to choose how they want to deal with the warning, what do you think?  @annatisch